### DISCUSSION TITLE
Using the output of a command as action for a remapped key?

### DISCUSSION THREAD

**Discussion by cmvizitiu** [sezanzeb/input-remapper#1008] [Q&A]
A few times a month I have to download some PDFs and file them with a certain name pattern. No, the site is definitely not under my control and they don't care. For years now I've used AutoKey's feature of running an external command and using the output of that command as input for a remapped key action.

Is there any way I can do the same with "Input-remapper"? What I need is a way to generate the string `YYYY-MM-DD__YYYY-MM-DD_ ` in an input field (the "Save As" dialog) with the actual dates in the pattern being calculated as the previous month's first day and last day e.g. 
```
#!/bin/bash
first_date=$(date -d "`date +%Y%m01` -1 month" +%Y-%m-%d)
last_date=$(date -d "`date +%Y%m01` -1 day" +%Y-%m-%d)
echo -e "${first_date}__${last_date}_
```
Maybe I don't need an external script for that but if that's the case I can't figure it out... 

Any ideas?

**Comment by iFabse** [comment]
So I am using Gnome and I used the built in keyboard shortcut and created my own. Just a random shortcut you don't use and execute a command - you can run your script - in input remapper, you can then  remap to your created shortcut
